src/EnergyPlus/Furnaces.cc

```diff
@@ -9445,7 +9716,59 @@ namespace Furnaces {
 			} else {
 				FanSpeedRatio = CompOnFlowRatio;
 			}
-		} else {
+		}
+		else if (true == Furnace(FurnaceNum).bIsIHP)
+		{
+			if (!CurDeadBandOrSetback(Furnace(FurnaceNum).ControlZoneNum) && present(SpeedNum)){
+			//if(present(SpeedNum)) {
+				CompOnMassFlow = GetAirMassFlowRateIHP(Furnace(FurnaceNum).CoolingCoilIndex, SpeedNum, SpeedRatio, false);
+				CompOnFlowRatio = CompOnMassFlow /
+					GetAirMassFlowRateIHP(Furnace(FurnaceNum).CoolingCoilIndex, GetMaxSpeedNumIHP(Furnace(FurnaceNum).CoolingCoilIndex), 1.0, false);
+				MSHPMassFlowRateLow = GetAirMassFlowRateIHP(Furnace(FurnaceNum).CoolingCoilIndex, SpeedNum, 0.0, false);
+				MSHPMassFlowRateHigh = GetAirMassFlowRateIHP(Furnace(FurnaceNum).CoolingCoilIndex, SpeedNum, 1.0, false);
+			}
+
+			// Set up fan flow rate during compressor off time
+			if (Furnace(FurnaceNum).OpMode == ContFanCycCoil && present(SpeedNum)) {
+				if (Furnace(FurnaceNum).AirFlowControl == UseCompressorOnFlow && CompOnMassFlow > 0.0) {
+					CompOffMassFlow = GetAirMassFlowRateIHP(Furnace(FurnaceNum).CoolingCoilIndex, SpeedNum, 1.0, false);
+					CompOffFlowRatio = CompOffMassFlow /
+						GetAirMassFlowRateIHP(Furnace(FurnaceNum).CoolingCoilIndex, GetMaxSpeedNumIHP(Furnace(FurnaceNum).CoolingCoilIndex), 1.0, false);
+				}
+			}
```

Comment:
the max speed will differ by individual working modes. So, GetMaxSpeedNumIHP(Furnace(FurnaceNum).CoolingCoilIndex) will check the current work mode first, and then, get the max speed. 

the air flow rate will differ by individual working modes (more than 5 modes) and speed levels. So, GetAirMassFlowRateIHP will check the current work mode and allowed speed first, and then, get the air flow rate. They can’t be contained in one array.